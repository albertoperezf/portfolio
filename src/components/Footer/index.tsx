import Link from "next/link";
import styles from "@/styles/Home.module.css";
import { useRouter } from "next/router";

const homeLinks = [
  { link: "/contact", subtitle: "Some ways to reach me.", title: "Contact" },
];
const contactLinks = [
  {
    link: "https://stackoverflow.com/users/6450931/alberto-perez",
    subtitle: `Find my interactions in the world's biggest developer platform.`,
    title: "Stackoverflow",
    target: "_blank",
  },
  {
    link: "https://www.linkedin.com/in/albertojpf/",
    subtitle: "Learn more about my professional trajectory.",
    title: "LinkedIn",
    target: "_blank",
  },
  {
    link: "https://github.com/albertoperezf",
    subtitle: "Check out some of my personal work (including this portfolio).",
    title: "Github",
    target: "_blank",
  },
  {
    link: "mailto:albertojperezf@gmail.com",
    subtitle: "Contact directly with me.",
    title: "Email",
    target: "_blank",
  },
];

export default function Footer(): JSX.Element {
  const router = useRouter();

  const generateLink = ({
    link,
    subtitle,
    target = "",
    title,
  }: {
    link: string;
    subtitle: string;
    target?: string;
    title: string;
  }): JSX.Element => (
    <Link
      key={title}
      href={link}
      className={styles.card}
      target={target}
      rel="noopener noreferrer"
    >
      <h2>
        {title} <span>-&gt;</span>
      </h2>
      <p>{subtitle}</p>
    </Link>
  );

  return (
    <div className={styles.grid}>
      {(router.pathname === "/contact" ? contactLinks : homeLinks).map((link) =>
        generateLink(link),
      )}
    </div>
  );
}
