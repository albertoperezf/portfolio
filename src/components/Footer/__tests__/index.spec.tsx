import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Footer from "..";

jest.mock("next/router", () => ({
  useRouter() {
    return {
      pathname: "/contact",
    };
  },
}));

describe("Footer component", () => {
  it("should render three links with correct hrefs and text", () => {
    render(<Footer />);
    const links = screen.getAllByRole("link");
    expect(links).toHaveLength(4);
    expect(links[0]).toHaveAttribute(
      "href",
      "https://stackoverflow.com/users/6450931/alberto-perez",
    );
    expect(links[0]).toHaveTextContent("Stackoverflow ->");
    expect(links[1]).toHaveAttribute(
      "href",
      "https://www.linkedin.com/in/albertojpf/",
    );
    expect(links[1]).toHaveTextContent("LinkedIn ->");
    expect(links[2]).toHaveAttribute(
      "href",
      "https://github.com/albertoperezf",
    );
    expect(links[2]).toHaveTextContent("Github ->");
    expect(links[3]).toHaveTextContent("Email ->");
  });

  it("should have correct rel attribute for links", () => {
    render(<Footer />);
    const links = screen.getAllByRole("link");
    links.forEach((link) => {
      expect(link).toHaveAttribute("rel", "noopener noreferrer");
    });
  });

  it("should have correct span element for arrow in links", () => {
    render(<Footer />);
    const arrows = screen.getAllByText("->");
    expect(arrows).toHaveLength(4);
    arrows.forEach((arrow) => {
      expect(arrow.tagName).toBe("SPAN");
    });
  });

  it("should match snapshot", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
