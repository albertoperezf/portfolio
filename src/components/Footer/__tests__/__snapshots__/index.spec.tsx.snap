// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should match snapshot 1`] = `
<div>
  <div
    class="grid"
  >
    <a
      class="card"
      href="https://stackoverflow.com/users/6450931/alberto-perez"
      rel="noopener noreferrer"
      target="_blank"
    >
      <h2>
        Stackoverflow
         
        <span>
          -&gt;
        </span>
      </h2>
      <p>
        Find my interactions in the world's biggest developer platform.
      </p>
    </a>
    <a
      class="card"
      href="https://www.linkedin.com/in/albertojpf/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <h2>
        LinkedIn
         
        <span>
          -&gt;
        </span>
      </h2>
      <p>
        Learn more about my professional trajectory.
      </p>
    </a>
    <a
      class="card"
      href="https://github.com/albertoperezf"
      rel="noopener noreferrer"
      target="_blank"
    >
      <h2>
        Github
         
        <span>
          -&gt;
        </span>
      </h2>
      <p>
        Check out some of my personal work (including this portfolio).
      </p>
    </a>
    <a
      class="card"
      href="mailto:albertojperezf@gmail.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <h2>
        Email
         
        <span>
          -&gt;
        </span>
      </h2>
      <p>
        Contact directly with me.
      </p>
    </a>
  </div>
</div>
`;
